### DISCUSSION TITLE
solarman_dtsd422-d3.yaml

### DISCUSSION THREAD

**Discussion by pmitrus** [davidrapan/ha-solarman#237] [General]
In solarman_dtsd422-d3.yaml there are voltage, current, power and frequency measurements avaiable but no any power ones. Device while presented on solarman web page also offers positive and negative energy. I'm considering if this is taken directly from device or calculated by sestem? Has anyone have information here?

**Comment by davidrapan** [comment]
Hi @pmitrus, they could be exposed using additional registers but due to limited availability of the docs for this device it's hard to determine them. Or those power values could be simply just calculated using voltage and current.

**Reply by pmitrus** [reply]
Thanks, that is true. But I suppose probing by device itself is more often and accurate than calculating this on server. And as I'm observing solarman web-page when I have wifi switched off, the energy measurement are delivered but delayed, so it means that meter is calculating this, and when transmission is restored all data are updated on solarman servers. The only case where not is when day is changed. Then I assume that solarman also is using "daily positive energy"  I'll try to search in registers outside already known maybe will find something.

**Reply by davidrapan** [reply]
You can try to contact them to provide you with list of registers. They are usually willing to help. 😉

If they give you the documentation don't forget to share it with us so I can put it on the wiki.
